### ISSUE TITLE
External Fact repmgr missing warning.

### ISSUE THREAD

**Issue by guardfox** [chrisjohntapp/puppet-repmgr#1]
Hello,
This bug affects  Non PostgreSQL nodes running Centos 7 agent  (puppet-agent-1.10.10-1.el7.x86_64)  , Puppet Master 4.8.2, in result running command "puppet agent -t" :

Warning: Facter: external fact file "/opt/puppetlabs/puppet/cache/facts.d/repmgrdStatus.sh" had output on stderr: which: no repmgr in (/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/opt/puppetlabs/bin:/root/bin:/sbin)
Warning: Facter: external fact file "/opt/puppetlabs/puppet/cache/facts.d/realPrimary.sh" had output on stderr: which: no locate in (/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/opt/puppetlabs/bin:/root/bin:/sbin)
Warning: Facter: external fact file "/opt/puppetlabs/puppet/cache/facts.d/replicationTier.sh" had output on stderr: which: no repmgr in (/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/opt/puppetlabs/bin:/root/bin:/sbin)

So basically  external fact is provisioned to all other nodes. 
I did not find a way to restrict that this facts are provisioned only to nodes wich use class "repmgr".
In order to mute this warrnings, I  would suggest adding in facts scripts following line before executing original code:
$(which repmgr &> /dev/null) ; rc=$?; if [[ $rc = 1 ]]; then exit 0; fi

Thank you.

**Comment by chrisjohntapp** [comment]
Thanks, that would do the trick; I'll add something similar shortly. possibly just
`which repmgr &>/dev/null || exit 0` 

I would like to write these as native ruby facts and might have some free time to do so soon; then I could use confine.

**Comment by chrisjohntapp** [comment]
Committed.
